test/unit/tags/comment_tag_unit_test.rb

```diff
@@ -31,25 +31,45 @@ def test_does_not_parse_nodes_inside_a_comment
     LIQUID
   end
 
-  def test_allows_incomplete_tags_inside_a_comment
-    assert_template_result("", <<~LIQUID.chomp)
+  def test_allows_unclosed_tags
+    assert_template_result('', <<~LIQUID.chomp)
       {% comment %}
-        {% assign foo = "1"
+        {% if true %}
       {% endcomment %}
     LIQUID
+  end
 
-    assert_template_result("", <<~LIQUID.chomp)
+  def test_open_tags_in_comment
+    assert_template_result('', <<~LIQUID.chomp)
       {% comment %}
-        {% comment %}
-          {% invalid
-        {% endcomment %}
+        {% assign a = 123 {% comment %}
       {% endcomment %}
     LIQUID
 
-    assert_template_result("", <<~LIQUID.chomp)
-      {% comment %}
-      {% {{ {%- endcomment %}
-    LIQUID
+    assert_raises(Liquid::SyntaxError) do
+      assert_template_result("", <<~LIQUID.chomp)
+        {% comment %}
+          {% assign foo = "1"
+        {% endcomment %}
+      LIQUID
+    end
+
+    assert_raises(Liquid::SyntaxError) do
+      assert_template_result("", <<~LIQUID.chomp)
+        {% comment %}
+          {% comment %}
+            {% invalid
+          {% endcomment %}
+        {% endcomment %}
+      LIQUID
+    end
+
+    assert_raises(Liquid::SyntaxError) do
+      assert_template_result("", <<~LIQUID.chomp)
+        {% comment %}
+        {% {{ {%- endcomment %}
+      LIQUID
+    end
```

Comment:
I agree, however, we can't introduce a change that will break existing Liquid templates.

For an example,
```liquid
{% comment %}
  {% assign a = 123
  {% endcomment %}
{% endcomment %}
```
This Liquid code is a valid Liquid code with version `5.4.0` because the `assign` tag is captured as `{% assign a = 123 {% endcomment %}`. (Liquid allows extra string inside a tag)

So, if we parse `comment` tag like the `raw` tag, we would capture, we would ignore the `{% assign a = 123` and capture `{% endcomment %}` from line 3 as the comment tag's delimiter. Since the template has an extra `{% endcomment %}` tag delimiter, the template is no longer valid.